server/http/ipfs.go

```diff
@@ -125,8 +126,18 @@ func ipfsHandler(lassie *lassie.Lassie) func(http.ResponseWriter, *http.Request)
 		log.Debugw("fetching cid into car file", "cid", rootCid.String(), "file", carfile.Name())
 		id, _, err := lassie.Fetch(req.Context(), rootCid, linkSystem)
 		if err != nil {
-			logger.LogStatus(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch cid: %v", err))
-			res.WriteHeader(http.StatusInternalServerError)
+			var status int
+			var logMsg string
+			if err == retriever.ErrNoCandidates {
```

Comment:
```suggestion
			if errors.Is(err, retriever.ErrNoCandidates) {
```

I think it's going to be best to use `errors.Is` for this because we're doing a bit of wrapping in there and we don't want this to be brittle if/when this error gets wrapped too